### DISCUSSION TITLE
How could I implement a custom sorter with fzflua for Smart Open

### DISCUSSION THREAD

**Discussion by srcrip** [ibhagwan/fzf-lua#1619] [Q&A] [answered]
Hey there,

Check this out: https://www.reddit.com/r/neovim/comments/1hh44pd/is_there_an_smartopennvim_alternative_from_fzflua/

there's some interest in having this plugin updated to support fzflua. I tried to PR it but I wasn't sure how to implement a custom sorter in fzflua. Is this even possible?

This is the plugin in question: https://github.com/danielfalk/smart-open.nvim

**Comment by ibhagwan** [comment]
> how to implement a custom sorter in fzflua. Is this even possible?

If you mean how to items are sorted after they’re sent to fzf, that would be impossible with fzf-lua as this falls under the realm of fzf the binary, once fzf-lua sends the items to fzf there’s no control over the fuzzy matching, you can experiment with `--tiebreak` or `--scheme` (see `man fzf`) and get what you want?

**Reply by srcrip** [reply]
Yeah basically this plugin combined old files, files in fit, files you have open, etc etc into one big smart picker with an algorithm that tries to just sort by the most relevant to you. So there would be no way to make that work with fzflua?

**Reply by ibhagwan** [reply]
> Yeah basically this plugin combined old files, files in fit, files you have open, etc etc into one big smart picker with an algorithm that tries to just sort by the most relevant to you. So there would be no way to make that work with fzflua?

With some work you can combine all of these entries, this came up a few times in the issues and there was also one attempt of a user to make plugin for that but I can’t recall the name ATM.

At the end of the day anything is possible, just depends how much time you’re willing to invest in this as controlling the sorting is going to be non trivial for reasons I mentioned above, take a look at how zoxide `zi` controls the sorting, maybe you’ll get an idea from it?

**Reply by srcrip** [reply]
is the method described here a good example of doing a custom sort with fzf-lua? https://www.reddit.com/r/neovim/comments/1hmoa2z/comment/m3vkvba/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Reply by ibhagwan** [reply]
> is the method described here a good example of doing a custom sort with fzf-lua? https://www.reddit.com/r/neovim/comments/1hmoa2z/comment/m3vkvba/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

I don’t know if you can qualify this as custom sort, using fzf_live (or any “live” variant) you’re essentially disabling the fuzzy matching and turning fzf into a selector only and each key press generates a new command, now if the piped shell utilities in this case qualify as custom sort? Perhaps.

**Reply by ibhagwan** [reply, marked as answer]
See https://github.com/junegunn/fzf/issues/1213#issuecomment-365876248
